apps/cowswap-frontend/src/modules/appData/utils/buildAppData.ts

```diff
@@ -107,3 +111,20 @@ export async function updateHooksOnAppData(
     appDataKeccak256,
   }
 }
+
+function mergeHooks(
+  hooks1: OrderInteractionHooks | undefined,
+  hooks2: OrderInteractionHooks | undefined
+): OrderInteractionHooks | undefined {
+  if (!hooks1 || !hooks2) return undefined
```

Comment:
Should it be:

```suggestion
  if (!hooks1 && !hooks2) return undefined
```

Otherwise, if one of them is unset it'll always return `undefined`

Reply:
Very good catch! Thanks!